TourMate/TourMate/Frontend/User/Controllers/FirebaseUserController.swift

```diff
@@ -46,4 +46,27 @@ struct FirebaseUserController: UserController {
         return (userAdapter.toUser(adaptedUser: adaptedUser), errorMessage)
     }
 
+    func getUser(email: String) async -> (User?, String) {
+        guard Auth.auth().currentUser != nil else {
+            return (nil, Constants.messageUserNotLoggedIn)
+        }
+
+        let (adaptedUsers, errorMessage) = await firebasePersistenceManager.fetchItems(field: "email", isEqualTo: email)
+
+        guard errorMessage.isEmpty else {
+            return (nil, errorMessage)
+        }
+
+        // should not have more than 1 user with the same email
+        guard adaptedUsers.count <= 1 else {
+            preconditionFailure()
+        }
+
+        guard let adaptedUser = adaptedUsers.first as? FirebaseAdaptedUser else {
+            return (nil, "") // user not found
+        }
```

Comment:
I think we can pass in the errorMessage?
```suggestion
        guard let adaptedUser = adaptedUsers.first as? FirebaseAdaptedUser else {
            return (nil, "User not found") // user not found
        }
```

Reply:
If we pass in the errorMessage, the front-end will register this as an error. Then it will display the error page instead. It's like that time we mentioned like (nil, "") means no item found in db?

Reply:
Right, I forgot about that 👌